ovve_ui/utils/params.py

```diff
@@ -5,34 +5,76 @@
 from typing import Union
 
 
+# Params defined in serialpacketv0.26.pptx
 class Params():
     def __init__(self) -> None:
+        self.seq_num: int = 0
+        self.packet_version: int = 0
+        self.mode: int = 0
+        self.resp_rate_meas: int = 0
+        self.resp_rate_set: int = 0
+        self.tv_meas: int = 0
+        self.tv_set: int = 0
+        self.ie_ratio_meas: int = 0
+        self.ie_ratio_set: int = 0
         self.peep: int = 0
-        self.tv_insp: int = 0
-        self.tv_exp: int = 0
         self.ppeak: int = 0
         self.pplat: Union[int, float] = 0
+        self.pressure: Union[int, float] = 0
+        self.flow: Union[int, float] = 0
+        self.tv_insp: int = 0
+        self.tv_exp: int = 0
+        self.tv_min: int = 0 
+        self.control_state: int = 0
+        self.battery_level: int = 0
+        #TODO: handle alarms
 
-    def set_test_params(self) -> None:
-        self.peep = 5
-        self.tv_insp = 435
-        self.tv_exp = 340
-        self.ppeak = 20
-        self.pplat = 2.5
-
+   
     def to_JSON(self) -> str:
         j = {}
+        j['seq_num'] = self.seq_num
+        j['packet_version'] = self.packet_version
+        j['mode'] = self.mode
+        j['rep_rate_meas'] = self.resp_rate_meas
+        j['resp_rate_set'] = self.resp_rate_set
+        j['tv_meas'] = self.tv_meas
+        j['tv_set'] = self.tv_set
+        j['ie_ratio_meas'] = self.ie_ratio_meas
+        j['ie_ratio_set'] = self.ie_ratio_set
         j['peep'] = self.peep
-        j['tv_insp'] = self.tv_insp
-        j['tv_exp'] = self.tv_exp
         j['ppeak'] = self.ppeak
         j['pplat'] = self.pplat
+        j['pressure'] = self.pressure
+        j['flow'] = self.flow
+        j['tv_insp'] = self.tv_insp
+        j['tv_exp'] = self.tv_exp
+        j['tv_min'] = self.tv_min 
+        j['control_state'] = self.control_state
+        j['battery_level'] = self.battery_level
         return json.dumps(j)
 
-    def from_dict(self, params_dict: dict) -> None:
-        self.peep = params_dict['peep']
-        self.tv_insp = params_dict['tv_insp']
-        self.tv_exp = params_dict['tv_exp']
-        self.ppeak = params_dict['ppeak']
-        self.pplat = params_dict['pplat']
-        
\ No newline at end of file
+    def from_dict(self, j: dict) -> None:
```

Comment:
We may want to add a means of error handling with this, but it might involve making an attribute creating the keys we are actually looking for and returning `None` if a name isn't coming out right.

Reply:
I agree - populating dictionaries can be error prone.  I'm not sure how to implement what you mean.  Can you add it in my branch?

Reply:
Yeah I can try and figure it out and add something.

Reply:
I put a todo tag on both dictionary conversions.  Let's address it later.